src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && prey.hidden === false) {
+      prey.health -= 50;
+
+      Animal.alive = Animal.alive.filter(animal => animal.health !== 0);
```

Comment:
This condition is better in case bite would have different number.
```suggestion
      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```